block/manager.go

```diff
@@ -832,7 +829,13 @@ func (m *Manager) recordMetrics(block *types.Block) {
 func (m *Manager) submitBlocksToDA(ctx context.Context) error {
 	submittedAllBlocks := false
 	backoff := initialBackoff
-	blocksToSubmit := m.pendingBlocks.getPendingBlocks()
+	blocksToSubmit, err := m.pendingBlocks.getPendingBlocks(ctx)
+	if len(blocksToSubmit) == 0 {
+		return err
+	}
+	if err != nil {
+		m.logger.Error("error while fetching blocks pending DA", "err", err)
+	}
```

Comment:
OK, this is a bit hacky, because it handles following situations:
 - no blocks and no error is returned ("something is wrong")
 - no blocks and an error is returned ("all blocks are already submitted to DA - normal case")
 - some blocks and an error is returned ("something failed during the call but we want to submit as much blocks as possible")
 - some blocks and no error is returned ("normal case")

Reply:
Edited/updated my first answer. Hope this is fine. There is also a comment in the code.